unit test failure: FAIL: TestExpandPath

Test failure when running using centos7 and opensuse 15.0.

Git blame says @ikaneshiro last edited, so let's start with assigning there.

The output is very hard for me to understand and it appears there are inline words for what should be character escape sequences? (tab, newline, space).

@ikaneshiro Let me know if you'd like me to try something or experiment with it myself. I may have some cycles at the end of next week.

```
    --- FAIL: TestExpandPath/PathWhitespace (0.00s)
        files_test.go:230: Generated 5 results: [/tmp/files-test-dir-061748190/dirL1 /tmp/files-test-dir-061748190/dir
            newline /tmp/files-test-dir-061748190/dir space /tmp/files-test-dir-061748190/dir   tab /tmp/files-test-dir-061748190/file]
        files_test.go:231: Correct 5 results: [/tmp/files-test-dir-061748190/dir        tab /tmp/files-test-dir-061748190/dir
            newline /tmp/files-test-dir-061748190/dir space /tmp/files-test-dir-061748190/dirL1 /tmp/files-test-dir-061748190/file]
        files_test.go:232: matched files are not correct
    --- FAIL: TestExpandPath/PathAndFilenameWhitespace (0.00s)
        files_test.go:230: Generated 14 results: [/tmp/files-test-dir-061748190/dirL1/dirL2 /tmp/files-test-dir-061748190/dirL1/file /tmp/files-test-dir-061748190/dir
            newline/file /tmp/files-test-dir-061748190/dir
            newline/file
            newline /tmp/files-test-dir-061748190/dir
            newline/file space /tmp/files-test-dir-061748190/dir
            newline/file        tab /tmp/files-test-dir-061748190/dir space/file /tmp/files-test-dir-061748190/dir space/file
            newline /tmp/files-test-dir-061748190/dir space/file space /tmp/files-test-dir-061748190/dir space/file     tab /tmp/files-test-dir-061748190/dir   tab/file /tmp/files-test-dir-061748190/dir      tab/file
            newline /tmp/files-test-dir-061748190/dir   tab/file space /tmp/files-test-dir-061748190/dir        tab/file        tab]
        files_test.go:231: Correct 14 results: [/tmp/files-test-dir-061748190/dir       tab/file /tmp/files-test-dir-061748190/dir      tab/file        tab /tmp/files-test-dir-061748190/dir   tab/file
            newline /tmp/files-test-dir-061748190/dir   tab/file space /tmp/files-test-dir-061748190/dir
            newline/file /tmp/files-test-dir-061748190/dir
            newline/file        tab /tmp/files-test-dir-061748190/dir
            newline/file
            newline /tmp/files-test-dir-061748190/dir
            newline/file space /tmp/files-test-dir-061748190/dir space/file /tmp/files-test-dir-061748190/dir space/file        tab /tmp/files-test-dir-061748190/dir space/file
            newline /tmp/files-test-dir-061748190/dir space/file space /tmp/files-test-dir-061748190/dirL1/dirL2 /tmp/files-test-dir-061748190/dirL1/file]
        files_test.go:232: matched files are not correct
```

--- a/internal/pkg/build/files/files_test.go
+++ b/internal/pkg/build/files/files_test.go
@@ -10,7 +10,7 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
-	"reflect"
+	"strconv"
 	"testing"
 )
 
@@ -58,6 +58,22 @@ files-test-dir/
 ├── file
 └── .file
 */
+
+func formatSlice(slice []string) (s string) {
+	for _, elm := range slice {
+		s += strconv.Quote(elm) + "\n"
+	}
+	return s
+}
+
+func contains(slice []string, s string) bool {
+	for _, elm := range slice {
+		if elm == s {
+			return true
+		}
+	}
+	return false
+}
 
 func createTestDirLayout(t *testing.T) string {
 	testDirName, err := ioutil.TempDir("", "files-test-dir-")
@@ -226,10 +242,13 @@ func TestExpandPath(t *testing.T) {
 				correct = append(correct, testDir+"/"+c)
 			}
 
-			if !reflect.DeepEqual(files, correct) {
-				t.Logf("Generated %d results: %s", len(files), files)
-				t.Logf("Correct %d results: %s", len(correct), correct)
-				t.Errorf("matched files are not correct")
+			for _, c := range correct {
+				if !contains(files, c) {
+					t.Logf("Generated %d results: %s", len(files), formatSlice(files))
+					t.Logf("Correct %d results: %s", len(correct), formatSlice(correct))
+					t.Errorf("matched files are not correct")
+					break
+				}
 			}
 		})
 	}

